DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/vlan/test_all_supported_modes.py

```diff
@@ -0,0 +1,286 @@
+
+    # 5. Send traffic to rx_ports.
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(5)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    # 5.1 Verify traffic.
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    ti_to_rx_port_map = get_traffic_port_vlan_mapping(streams, port_map, tg_ports)
+    for row in stats.Rows:
+        if row["Rx Port"] in ti_to_rx_port_map[row["Traffic Item"]]:
+            assert tgen_utils_get_loss(row) == 0.000, \
+                f'Traffic leak for traffic item: {row["Traffic Item"]} on port {row["Rx Port"]}'
+        else:
+            assert tgen_utils_get_loss(row) == 100.000, \
+                f'No traffic for traffic item : {row["Traffic Item"]} on port {row["Rx Port"]}'
+    await tgen_utils_stop_protocols(tgen_dev)
+
+
+@pytest.mark.asyncio
+@pytest.mark.usefixtures("cleanup_bridges")
+async def test_vlan_all_supported_modes_unicast(testbed):
+    """
+    Test Name: Unicast with all modes supported
+    Test Suite: suite_functional_vlan
+    Test Overview: Test VLAN with unicast when all modes supported
+    Test Procedure:
+    1.Initiate test params.
+    2.Set links to vlans.
+    # 3.Map receiving and non receiving dut_ports.
+        Port 1 ->  tx_port
+        Port 2 ->  rx_port
+    4.Setup packet stream(s) for the broadcast packet:
+    5.Send traffic to rx_ports
+            5.1 Verify traffic
+    """
+
+    # 1. Initiate test params.
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgn connections")
+        return
+    device = dent_devices[0].host_name
+    tg_ports = tgen_dev.links_dict[device][0]
+    dut_ports = tgen_dev.links_dict[device][1]
+    traffic_duration = 5
+
+    await configure_bridge_setup(device, dut_ports)
+
+    # 2.Set links to vlans.
+    await configure_vlan_setup(device, port_map, dut_ports)
+
+    # 3.Map receiving and non receiving dut_ports.
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "100.1.1.2", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[1], "ip": "100.1.1.3", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[2], "ip": "100.1.1.4", "gw": "100.1.1.6", "plen": 24, },
+         {"ixp": tg_ports[3], "ip": "100.1.1.5", "gw": "100.1.1.6", "plen": 24, }])
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+
+    tx_ports = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[x]][0]["name"] for x in range(1, 4)]
+
+    mac = {tg_ports[idx]: f"02:00:00:00:00:0{idx + 1}" for idx in range(4)}
+
+    vlan_to_mac = {2: "02:00:00:00:00:02",
+                   3: "02:00:00:00:00:03",
+                   4: "02:00:00:00:00:04",
+                   "X": "02:00:00:00:00:04",
+                   1: "02:00:00:00:00:04", }
```

Comment:
You already have mac addrs in `mac` dict:
```python
vlan_to_mac = {
    2: mac[tg_ports[1]],
    3: mac[tg_ports[2]],
    4: mac[tg_ports[3]],
    "X": mac[tg_ports[3]],
    1: mac[tg_ports[3]],
}
```

Reply:
i believe this needed to distinguish from which port to send packet with certain vlan